### ISSUE TITLE
Too many open files in Stat_Worker

### ISSUE THREAD

**Issue by felskrone** [felskrone/salt-eventsd#40]
Do we properly close all handles where necessary?

```bash
2015-03-08 13:22:43 [Stat_Worker                             ][ERROR   ] 23121378# Connecting to the mysql-server failed:
2015-03-08 13:22:43 [Stat_Worker                             ][ERROR   ] (2005, "Unknown MySQL server host 'custom_hostname' (24)")
2015-03-08 13:22:53 [salteventsd.worker                      ][INFO    ] 23121379# started
2015-03-08 13:22:53 [salteventsd.daemon                      ][CRITICAL] Failed to write state to /var/run/salt-eventsd.status
2015-03-08 13:22:53 [Stat_Worker                             ][ERROR   ] 23121379# Connecting to the mysql-server failed:
2015-03-08 13:22:53 [salteventsd.daemon                      ][ERROR   ] [Errno 24] Too many open files: '/var/run/salt-eventsd.status'
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.7/salteventsd/daemon.py", line 516, in _write_state
IOError: [Errno 24] Too many open files: '/var/run/salt-eventsd.status'
```

**Comment by felskrone** [comment]
python    23708 23714        root   13u     IPv4          181368870         0t0        TCP server01.intern:54561->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   14u     IPv4          181357966         0t0        TCP server01.intern:54566->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   15u     IPv4          181369256         0t0        TCP server01.intern:54567->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   16u     IPv4          181369391         0t0        TCP server01.intern:54568->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   17u     IPv4          181369562         0t0        TCP server01.intern:54571->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   18u     IPv4          181369673         0t0        TCP server01.intern:54572->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   19u     IPv4          181365262         0t0        TCP server01.intern:54573->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   20u     IPv4          181365302         0t0        TCP server01.intern:54576->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   21u     IPv4          181370182         0t0        TCP server01.intern:54577->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   22u     IPv4          181370468         0t0        TCP server01.intern:54581->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   23u     IPv4          181365531         0t0        TCP server01.intern:54582->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   24u     IPv4          181371194         0t0        TCP server01.intern:54588->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   25u     IPv4          181370729         0t0        TCP server01.intern:54583->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   26u     IPv4          181358148         0t0        TCP server01.intern:54586->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   27u     IPv4          181365693         0t0        TCP server01.intern:54587->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   28u     IPv4          181371322         0t0        TCP server01.intern:54591->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   29u     IPv4          181375899         0t0        TCP server01.intern:54609->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   30u     IPv4          181372003         0t0        TCP server01.intern:54592->api01.intern:mysql (CLOSE_WAIT)
python    23708 23714        root   31u     IPv4          181371639         0t0        TCP server01.intern:54593->api01.intern:mysql (CLOSE_WAIT)

**Comment by felskrone** [comment]
fixed in 0.9.3
